bamboolean/walkers/normalize.py

```diff
@@ -0,0 +1,77 @@
+from typing import TypeVar, Dict
+from bamboolean.node_visitor import NodeVisitor
+from bamboolean.ast import AST, Constraint, BinOp, UnaryOp, Bool
+from bamboolean.lexer import Token
+from bamboolean.tokens import NOT
+
+
+T1 = TypeVar('T1')
+T2 = TypeVar('T2')
+
+
+def flip_dict(d: Dict[T1, T2]) -> Dict[T2, T1]:
+    return {v: k for k, v in d.items()}
+
+
+def complementary_op(op):
+    flip_op = {
+        ('EQ', '=='): ('NE', '!='),
+        ('LT', '<'): ('GTE', '>='),
+        ('LTE', '<='): ('GT', '>'),
+        ('OR', 'OR'): ('AND', 'AND')
```

Comment:
These being strings is kinda meh, especially as some are used in the parser and lexer too. Can you make typing & reuse stronger here?

Reply:
Agreed, done